### PR TITLE
Add Factory 2.0 APIs For Branch SLAs and EOLs

### PR DESCRIPTION
**Pull request by mprahl**
This pull request adds three new APIs that enable us to define a package's branch and its corresponding SLAs (each having an EOL).

For more information, I have made a video demo on the changes:
https://fedorapeople.org/groups/factory2/sprint-014/mprahl-pdc-sla-apis.mp4

### CODE REVIEW

File: pdc/apps/componentbranch/filters.py [hunk 1]
```diff
@@ -5,19 +5,83 @@
 #
 
 import django_filters
+from django.db.models import Q
```
Comment (ralphbean) [inline pdc/apps/componentbranch/filters.py]: https://en.wikipedia.org/wiki/Q_(James_Bond)

File: pdc/apps/componentbranch/filters.py [hunk 2]
```diff
@@ -5,19 +5,83 @@
...
+            return False
+        else:
+            raise ValueError('"{0}" is not a valid bool'.format(item))
+
+
+def filter_active(queryset, value):
+    if not value:
+        return queryset
+    try:
+        proccess_value = string_to_bool(value)
```
Comment (ralphbean) [inline pdc/apps/componentbranch/filters.py]: I think "proccess" is spelled "process"?

Comment (mprahl) [inline pdc/apps/componentbranch/filters.py]: Embarrassing :smile:

File: pdc/apps/componentbranch/models.py [hunk 3]
```diff
@@ -35,7 +34,6 @@ def export(self):
             'slas': [{'name': sla_to_branch.sla.name,
                       'eol': sla_to_branch.eol.strftime('%Y-%m-%d')}
                      for sla_to_branch in self.slas.all()],
-            'active': self.active,
```
Comment (ralphbean) [inline pdc/apps/componentbranch/models.py]: This should probably still be exported, even if it is not kept as a database member?

That way a query can quickly tell "is this thing active"?

File: pdc/apps/componentbranch/filters.py (hunk 1 above)
Comment (mprahl) [inline pdc/apps/componentbranch/filters.py]: :gun:

File: pdc/apps/componentbranch/models.py (hunk 3 above)
Comment (ralphbean) [inline pdc/apps/componentbranch/models.py]: Oh -- Maybe that is handled by the "serializer" and not here.

Comment (mprahl) [inline pdc/apps/componentbranch/models.py]: Yes, that is handled by the serializer.

**Comment by mprahl** [comment]
I rebased the PR and resolved merge conflicts.

**Comment by hluk** [comment]
+1 (BTW, thanks for nice explanation in the video.)

**Comment by mprahl** [comment]
@hluk thanks for reviewing! Is this good to merge?

File: pdc/apps/componentbranch/filters.py [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+import django_filters
+from django.db.models import Q
+from datetime import datetime
+
+from pdc.apps.common.filters import CaseInsensitiveBooleanFilter
+from pdc.apps.componentbranch.models import (
+    ComponentBranch, SLA, SLAToComponentBranch)
+
+
+def string_to_bool(item):
```
Comment (simozhan) [inline pdc/apps/componentbranch/filters.py]: We have a similar method convert_str_to_bool in file apps/common/hacks.py
Maybe we can think about to merge them?

Comment (hluk) [inline pdc/apps/componentbranch/filters.py]: I think we can change it later. Would it be OK to merge now?

Comment (simozhan) [inline pdc/apps/componentbranch/filters.py]: @hluk sure. let's merge it now and think to change it in other commit.

**Pull request merged**